src/main/java/ch/njol/skript/Skript.java

```diff
@@ -1386,15 +1387,24 @@ public boolean check(final @Nullable ExpressionInfo<?, ?> i) {
 		return r;
 	}
 
-	@SuppressWarnings("unchecked")
 	public static <E extends Structure> void registerStructure(Class<E> c, String... patterns) {
+		if (PreloadingStructure.class.isAssignableFrom(c))
+			throw new IllegalArgumentException("Attempted to register a preloading structure using registerStructure instead of registerPreloadingStructure");
+
 		checkAcceptRegistrations();
 		String originClassPath = Thread.currentThread().getStackTrace()[2].getClassName();
 		SyntaxElementInfo<E> structureInfo = new SyntaxElementInfo<>(patterns, c, originClassPath);
-		if (PreloadingStructure.class.isAssignableFrom(c))
-			preloadingStructures.add((SyntaxElementInfo<? extends PreloadingStructure>) structureInfo);
-		else
-			normalStructures.add(structureInfo);
+		normalStructures.add(structureInfo);
+	}
+
+	public static <E extends PreloadingStructure> void registerPreloadingStructure(Class<E> c, int priority, String... patterns) {
```

Comment:
The method name should not be different, the only difference is that it accepts a priority, which you can assign a default value.
The methods would then become:
- `registerStructure(Class<E>, String...)` calling the next method with a default priority
- `registerStructure(Class<E>, int, String...)`